File: java/mix/C03GcdType1A.java
public class C03GcdType1A {
    public static int gcd(int a, int b) {
        a = Math.abs(a); b = Math.abs(b);
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }
}

File: test/cpp/mix/c03_gcd_type1a.cpp
#include <iostream>
#include <cmath>
using namespace std;

int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return abs(a);
}

int main() {
    cout << gcd(48, 18) << endl;
    return 0;
}
